### PR TITLE
[Doris On ES][Bug-Fix] ES queries always route at same 3 BE nodes (#4351)

### LINKED ISSUE #4351: [Doris On ES][Bug] ES queries always route at same 3 BE nodes
I create a external ES table and test some queries on ES.Then I find all of the queries route at same 3 BE nodes.
I find the code at fe/fe-core/src/main/java/org/apache/doris/planner/EsScanNode.java
227: candidateBeList.add(backendList.get(beIndex++ % numBe));
numBe is always 3 when number of be is greater than 3. So that candidateBeList always add 3 front of beckendList.
It should be corrected like:
candidateBeList.add(backendList.get(beIndex++ % backendList.size()));
Then FE can balance ES queries between BE nodes.

### PR DESCRIPTION
**Pull request by caoyang10**
Fix Issue:
https://github.com/apache/incubator-doris/issues/4351

### CODE REVIEW

File: fe/fe-core/src/main/java/org/apache/doris/planner/EsScanNode.java [hunk 1]
```diff
@@ -224,7 +224,7 @@ private void assignBackends() throws UserException {
                 List<Backend> candidateBeList = Lists.newArrayList();
                 if (usingRandomBackend) {
                     for (int i = 0; i < numBe; ++i) {
-                        candidateBeList.add(backendList.get(beIndex++ % numBe));
+                        candidateBeList.add(backendList.get(beIndex++ % backendList.size()));
```
Comment (wuyunfeng) [inline fe/fe-core/src/main/java/org/apache/doris/planner/EsScanNode.java:227]:
```suggestion
                        candidateBeList.add(backendList.get(beIndex++ % size));
```
obtain size = backendList.size() in other place

**Review by wuyunfeng — COMMENTED** [review]
This Looks good to me, but I left some minor comment

**Comment by caoyang10** [comment]
I've obtained size = backendList.size() in the front of loop. Is it better?
please check the commit 6e17409

**Review by wuyunfeng — APPROVED** [review]
It would be nice to add some UT to cover this case, but writing this is a bit complicated for you. At this time I would  +1 and  merge this PR soon. Thanks.

**Pull request merged**